packages/react/src/utils/useMnemonics.ts

```diff
@@ -0,0 +1,127 @@
+import type { MutableRefObject, RefObject } from 'react';
+import type { ElementOrRef } from '../types';
+import { useRef, useEffect } from 'react';
+import focusable from 'focusable';
+
+type useMnemonicsOptions = {
+  /**
+   * The container element or ref to use for matching elements for mnemonics.
+   * If not provided, the hook will return a ref that should be attached to a container.
+   */
+  elementOrRef?: ElementOrRef<HTMLElement>;
+
+  /**
+   * CSS selector to match for elements, defaults to focusable descendants.
+   */
+  matchingElementsSelector?: string;
+
+  /**
+   * Callback fired when a matching element is found via mnemonic keyboard entry.
+   */
+  onMatch: (element: HTMLElement) => void;
+
+  /**
+   * Whether mnemonic navigation is enabled. Defaults to true.
+   */
+  enabled?: boolean;
+};
+
+type useMnemonicsResults<T extends HTMLElement> = RefObject<T>;
```

Comment:
`useMnemonics` returns `containerRef`, which is a `MutableRefObject`. Should this be updated here?

Reply:
I think I was confused because it gets cast as a `MutableRefObject` in the `ActionList`.